src/test/java/apicurito/tests/steps/VerificationSteps/OperationVerifications.java

```diff
@@ -0,0 +1,248 @@
+        String of = OperationUtils.getOperationRoot().$(OperationSteps.REQUEST_BODY_SECTION).$(CommonUtils.DropdownButtons.PROPERTY_TYPE_OF.getButtonId()).getText();
+        collector.assertThat(of).as("Type of is %s but should be %s", of, expectedOf).isEqualTo(expectedOf);
+    }
+
+    @Then("^check that request body type as is \"([^\"]*)\"$")
+    public void checkThatRequestBodyTypeAsIs(String expectedAs) {
+        String as = OperationUtils.getOperationRoot().$(OperationSteps.REQUEST_BODY_SECTION).$(CommonUtils.DropdownButtons.PROPERTY_TYPE_AS.getButtonId()).getText();
+        collector.assertThat(as).as("Type as is %s but should be %s", as, expectedAs).isEqualTo(expectedAs);
+    }
+
+    @Then("^check that exist request form data$")
+    public void checkThatExistRequestFormData(DataTable table) {
+        CommonUtils.openCollapsedSection(OperationSteps.REQUEST_BODY_SECTION);
+
+        for (List<String> dataRow : table.cells()) {
+            checkRow(dataRow, OperationSteps.REQUEST_BODY_SECTION);
+        }
+    }
+
+    @Then("^check that exist query parameters$")
+    public void checkThatExistQueryParameter(DataTable table) {
+        CommonUtils.openCollapsedSection(OperationSteps.QUERY_PARAM_SECTION);
+
+        for (List<String> dataRow : table.cells()) {
+            checkRow(dataRow, OperationSteps.QUERY_PARAM_SECTION);
+        }
+    }
+
+    @Then("^check that exist header parameters$")
+    public void checkThatExistHeaderParameters(DataTable table) {
+        CommonUtils.openCollapsedSection(OperationSteps.HEADER_PARAM_SECTION);
+
+        for (List<String> dataRow : table.cells()) {
+            checkRow(dataRow, OperationSteps.HEADER_PARAM_SECTION);
+        }
+    }
+
+    private void checkRow(List<String> dataRow, By section) {
+        String rowType;
+        String message;
+
+        switch (section.toString()) {       //TODO find something better
+            case "By.cssSelector: requestbody-section":
+                rowType = "formdata-param-row";
+                message = "Form data parameter";
+                break;
+            case "By.cssSelector: query-params-section":
+                rowType = "query-param-row";
+                message = "Query parameter";
+                break;
+            case "By.cssSelector: header-params-section":
+                rowType = "header-param-row";
+                message = "Header parameter";
+                break;
+            default:
+                rowType = null;
+                message = null;
+        }
+
+        ElementsCollection queryRows = OperationUtils.getOperationRoot().$(section).$$(rowType);
+
+        if (queryRows.size() > 0) {
+            ElementsCollection names = OperationUtils.getOperationRoot().$(section).$$(By.className("name")).filter((text(dataRow.get(0))));
+
+            if (names.size() == 0) {
+                collector.fail("%s with name %s is not created!", message, dataRow.get(0));
+            } else {
```

Comment:
Is there a way how to break all these if statements in here to different helper methods? Or perhaps some good generic method?